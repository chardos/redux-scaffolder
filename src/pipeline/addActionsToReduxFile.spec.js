const { prettify } = require("../utils");
const addActionsToReduxFile = require("./addActionsToReduxFile");
const generateActionCode = require("../renderers/redux/generateActionCode");
const generateReducerCode = require("../renderers/redux/generateReducerCode");
const generateDuckCode = require("../renderers/redux/generateDuckCode");

describe("addActionsToReduxFile()", () => {
  describe("on an action file", () => {
    it("Matches the snapshot", () => {
      const existingFile = generateActionCode("Zone", ["AddZone"]);
      const result = addActionsToReduxFile(
        "Zone",
        ["RemoveZone"],
        existingFile
      );

      expect(prettify(result)).toMatchSnapshot();
    });
  });
  describe("on a reducer file", () => {
    it("Matches the snapshot", () => {
      const existingFile = generateReducerCode("Zone", ["AddZone"]);
      const result = addActionsToReduxFile(
        "Zone",
        ["RemoveZone"],
        existingFile
      );

      expect(prettify(result)).toMatchSnapshot();
    });
  });
  describe("on a duck file", () => {
    it("Matches the snapshot", () => {
      const existingFile = generateDuckCode("Zone", ["AddZone"]);
      const result = addActionsToReduxFile(
        "Zone",
        ["RemoveZone"],
        existingFile
      );

      expect(prettify(result)).toMatchSnapshot();
    });
  });
});
