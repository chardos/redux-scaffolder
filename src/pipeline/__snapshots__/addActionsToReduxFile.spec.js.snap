// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`addActionsToReduxFile() on a duck file Matches the snapshot 1`] = `
"export const ADD_ZONE = \\"ADD_ZONE\\";
export const REMOVE_ZONE = \\"REMOVE_ZONE\\";

export default function Zone(state = {}, action) {
  switch (action.type) {
    case ADD_ZONE:
      return state;

    case REMOVE_ZONE:
      return state;

    default:
      return state;
  }
}
export const addZone = () => {
  return {
    type: ADD_ZONE
  };
};
export const removeZone = () => {
  return {
    type: REMOVE_ZONE
  };
};
"
`;

exports[`addActionsToReduxFile() on a reducer file Matches the snapshot 1`] = `
"import { ADD_ZONE, REMOVE_ZONE } from \\"../actions/Zone\\";
export default function Zone(state = {}, action) {
  switch (action.type) {
    case ADD_ZONE:
      return state;

    case REMOVE_ZONE:
      return state;

    default:
      return state;
  }
}
"
`;

exports[`addActionsToReduxFile() on an action file Matches the snapshot 1`] = `
"export const ADD_ZONE = \\"ADD_ZONE\\";
export const REMOVE_ZONE = \\"REMOVE_ZONE\\";

export const addZone = () => {
  return {
    type: ADD_ZONE
  };
};
export const removeZone = () => {
  return {
    type: REMOVE_ZONE
  };
};
"
`;
